lib/ocrypto/ec_key_pair.go

```diff
@@ -0,0 +1,103 @@
+package ocrypto
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+)
+
+type eccMode uint8
+
+const (
+	eccModeSecp256r1 eccMode = 0
+	eccModeSecp384r1 eccMode = 1
+	eccModeSecp521r1 eccMode = 2
+	eccModeSecp256k1 eccMode = 3
+)
+
+type EcKeyPair struct {
+	privateKey *ecdsa.PrivateKey
+}
+
+// NewECKeyPair Generates an EC key pair of the given bit size.
+func NewECKeyPair(mode eccMode) (EcKeyPair, error) {
+
+	var c elliptic.Curve
+	switch mode {
+	case eccModeSecp256r1:
+		c = elliptic.P256()
+		break
```

Comment:
You don't need these breaks in go

https://go.dev/tour/flowcontrol/9

Reply:
Fixed